src/test/java/com/example/onlinebookstore/controller/ShoppingCartControllerTest.java

```diff
@@ -0,0 +1,215 @@
+import com.example.onlinebookstore.dto.cartitem.UpdateQuantityRequestDto;
+import com.example.onlinebookstore.repository.CartItemRepository;
+import com.example.onlinebookstore.security.JwtUtil;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.List;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+class ShoppingCartControllerTest {
+    protected static MockMvc mockMvc;
+    @Autowired
+    private ObjectMapper objectMapper;
+    @Autowired
+    private JwtUtil jwtUtil;
+    @Autowired
+    private CartItemRepository cartItemRepository;
+
+    @BeforeAll
+    static void setUp(@Autowired WebApplicationContext applicationContext) {
+        mockMvc = MockMvcBuilders
+                .webAppContextSetup(applicationContext)
+                .apply(springSecurity())
+                .build();
+    }
+
+    @Test
+    @DisplayName("Get user's cart")
+    @Sql(scripts = {
+            "classpath:database/books/add-users.sql",
+            "classpath:database/books/add-shopping_carts.sql",
+            "classpath:database/books/set-roles-to-users.sql"
+    }, executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+    @Sql(scripts = {
+            "classpath:database/books/clear-users_roles-table.sql",
+            "classpath:database/books/clear-shopping_carts-table.sql",
+            "classpath:database/books/clear-users-table.sql"
+    }, executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD)
+    public void getCart_ValidRequest_ShouldReturnCartResponseDto() throws Exception {
+        CartResponseDto expected = new CartResponseDto();
+        expected.setId(1L);
+        expected.setUserId(2L);
+
+        String token = getToken("second@user.com");
+        MvcResult result = mockMvc.perform(get("/api/cart")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .header(HttpHeaders.AUTHORIZATION, token))
+                .andExpect(status().isOk())
+                .andReturn();
+        CartResponseDto actual =
+                objectMapper.readValue(result.getResponse().getContentAsString(),
+                        CartResponseDto.class);
+
+        Assertions.assertEquals(expected.getId(), actual.getId());
+        Assertions.assertEquals(expected.getUserId(), actual.getUserId());
```

Comment:
use jsonpath function instead of objectMapper to reduce this code. 

e.g. .andExpect(jsonPath("$.blablabla").value("bla")
```suggestion
```